components/support/nimbus-fml/src/workflows.rs

```diff
@@ -158,6 +161,63 @@ pub(crate) fn fetch_file(files: &LoaderConfig, nm: &str) -> Result<()> {
     Ok(())
 }
 
+pub(crate) fn validate(cmd: &ValidateCmd) -> Result<()> {
+    let files: FileLoader = TryFrom::try_from(&cmd.loader)?;
+    let cwd = cmd.loader.cwd.to_str().unwrap();
+
+    let filename = &cmd.manifest;
+    let file_path = files.file_path(filename)?;
+    let parser: Parser = Parser::new(files, file_path.clone())?;
+    let mut loading = HashSet::new();
+    let manifest_front_end = parser.load_manifest(&file_path, &mut loading)?;
+
+    println!(
+        "Loaded modules:\n- {}\n",
+        loading
+            .iter()
+            .map(|id| id.to_string().replace(cwd, ""))
+            .collect::<Vec<String>>()
+            .join("\n- ")
+    );
+
+    println!("Validation manifest for different channels:");
+    let results = manifest_front_end
+        .channels
+        .iter()
+        .map(|c| {
+            let intermediate_representation = parser.get_intermediate_representation(c);
+            match intermediate_representation {
+                Ok(ir) => (c, ir.validate_manifest()),
+                Err(e) => (c, Err(e)),
+            }
+        })
+        .collect::<Vec<(&String, Result<_>)>>();
+
+    let mut errors = vec![];
+    for (channel, result) in results {
+        print!(r#"- "{}"..."#, channel);
+        match result {
+            Ok(_) => {
+                println!("valid ✓");
+            }
+            Err(e) => {
+                println!("invalid x");
+                println!("{}", e);
+                errors.push(e);
+            }
+        }
```

Comment:
Can we do a bit better with alignment of the statuses? e.g. 

```suggestion
        let status = match result {
            Ok(_) => "valid ✓",
            Err(e) => {
                println!("{}", e);
                errors.push(e);
                "invalid x"
            }
        };
        println!("- {channel:…<15}{status}");
```